ocs_ci/framework/pytest_customization/ocscilib.py

```diff
@@ -539,6 +540,48 @@ def process_cluster_cli_params(config):
     cluster_path = os.path.expanduser(cluster_path)
     if not os.path.exists(cluster_path):
         os.makedirs(cluster_path)
+
+    # create kubconfig if doesn't exists and OCP url and kubeadmin password is provided
+    kubeconfig_path = os.path.join(
+        cluster_path, ocsci_config.RUN["kubeconfig_location"]
+    )
+    if not os.path.isfile(kubeconfig_path) and not get_cli_param(
```

Comment:
should we do that for all clusters?

e.g. if ocsci_config.multicluster is set - we should iterate over all clusters to do this?

Reply:
I didn't tried it explicitly, but the `process_cluster_cli_params` function is called in `pytest_configure` [in following loop](https://github.com/red-hat-storage/ocs-ci/pull/11749/files#diff-81dd75ca7912048f20e991916771e11b16203494498849a39b6189401b3ac974L388-L393):
```python
    for i in range(ocsci_config.nclusters):
        log.info(f"Pytest configure switching to: cluster={i}")
        ocsci_config.switch_ctx(i)
        if not (config.getoption("--help") or config.getoption("collectonly")):
            process_cluster_cli_params(config)
```
So my understanding is that it should work out of the box.